### DISCUSSION TITLE
Only display chartModel X-values, without computing intermediate values

### DISCUSSION THREAD

**Discussion by Tboy70** [patrykandpatrick/vico#305] [Questions]
Hi, 

I would like to know if it's possible to display only the values of x-axis and not display the intermediate ones. 

For example , I have this : 

```
val chartEntryModel = entryModelOf(  
    mutableListOf<FloatEntry>().apply {  
	    add(FloatEntry(20f, 1f))  
	    add(FloatEntry(21f, 3f))  
	    add(FloatEntry(40f, 5f))  
	    }  
	)
```

On my chart, on my X-axis, I would like to have only 20, 21 and 40, and not display 22..39 (cause there is no time relation between x-axis values).

Thank you

**Comment by patrickmichalik** [comment]
Hello! As such, Vico’s charts are based on a standard Cartesian coordinate plane. However, you can normalize the _x_ values and use `AxisValueFormatter` to format the labels displayed along your `HorizontalAxis` instance (or instances). An example can be found [here](https://patrykandpatrick.com/vico/wiki/axes/#formatting-axis-values). In your case, a slightly more concise solution can be used:

```kotlin
val xValues = listOf(20f, 21f, 40f)
val yValues = listOf(1f, 3f, 5f)
val chartEntryModel = entryModelOf(yValues.mapIndexed { index, yValue -> entryOf(index, yValue) })
val horizontalAxisValueFormatter = AxisValueFormatter<AxisPosition.Horizontal> { value, _ ->
    String.format("%.0f", xValues[value.toInt()])
}
```
